datafusion/physical-expr/src/functions.rs

```diff
@@ -608,6 +608,27 @@ pub fn create_physical_fun(
                 _ => unreachable!(),
             },
         }),
+        BuiltinScalarFunction::RegexpLike => Arc::new(|args| match args[0].data_type() {
+            DataType::Utf8 => {
+                let func = invoke_on_array_if_regex_expressions_feature_flag!(
+                    regexp_like,
+                    i32,
+                    "regexp_like"
+                );
+                make_scalar_function_inner(func)(args)
+            }
+            DataType::LargeUtf8 => {
+                let func = invoke_on_array_if_regex_expressions_feature_flag!(
+                    regexp_like,
+                    i64,
+                    "regexp_like"
+                );
+                make_scalar_function_inner(func)(args)
+            }
+            other => {
+                internal_err!("Unsupported data type {other:?} for function regexp_like")
```

Comment:
```suggestion
                exec_err!("Unsupported data type {other:?} for function regexp_like")
```

Reply:
There is quite a few instances in the code that have this pattern (I was mirroring the existing regexp_... functions there) - I was going to create a followup PR that fixes them across the whole code base

Reply:
That would be most appreciated. Cleaning up internal errors is very good improvement